build-system/pr-check/dist-tests.js

```diff
@@ -29,21 +29,32 @@ const {
   startTimer,
   stopTimer,
   timedExecOrDie: timedExecOrDieBase,
+  timedExec: timedExecBase,
 } = require('./utils');
 const {determineBuildTargets} = require('./build-targets');
 const {isTravisPullRequestBuild} = require('../common/travis');
 
 const FILENAME = 'dist-tests.js';
 const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
 const timedExecOrDie = (cmd) => timedExecOrDieBase(cmd, FILENAME);
+const timedExec = (cmd) => timedExecBase(cmd, FILENAME);
 
 function main() {
   const startTime = startTimer(FILENAME, FILENAME);
 
-  if (!isTravisPullRequestBuild()) {
+  // NO SUBMIT PLZ
+  if (true || !isTravisPullRequestBuild()) {
     downloadDistOutput(FILENAME);
     timedExecOrDie('gulp update-packages');
-    timedExecOrDie('gulp integration --nobuild --headless --compiled');
+
+    const {status} = timedExec(
+      'gulp integration --nobuild --headless --compiled'
+    );
+
+    timedExecOrDie('gulp test-report-upload');
+    if (status) {
+      process.exit(status);
+    }
```

Comment:
It looks like the main goal here is to add a way to run command X, and if it fails, run command Y and then exit. Instead of implementing the same logic in multiple files in `build-system/pr-check`, I think we should add a function called `timedExecOrDieWithFinalize` (pick a more elegant name!), that takes a `cmd` and a `finalizeCmd`, and runs `finalizeCmd` whether or not `cmd` is successful. WDYT?

Reply:
I wonder if this might be over-abstracting a bit, and the calling pattern would be weird:

```javascript
timeExecOrDieWithFinalize('do command #1', 'gulp test-report-upload');
timeExecOrDieWithFinalize('do command #2', 'gulp test-report-upload');
timeExecOrDieWithFinalize('do command #3', 'gulp test-report-upload');
```

I feel like a more reusable variant that would fit this use case and others would be `timeExecOrThrow`, allowing calling code to use `try/catch/finally` to handle it. It would still require the calling code to exit (`catch (e) { process.exit(e.statusCode) }`) but might be more generally useful. WDYT?

Reply:
I fully agree with both your points: the `timedExecOrDieWithFinalize` pattern I suggested looks weird, and the `timeExecOrThrow` pattern you suggested is more elegant and usable. The new plan SGTM!

Reply:
Done, PTAL